AccessControl: Add FGAC to orgs endpoints

gamab opened the pull request:
**What this PR does / why we need it**:

> While in the first milestone we covered mostly all organization relevant endpoints with fine-grained permissions, there are few still missing the construction blocks.

In this first PR, I cover the following endpoints:
| Method | Url                        | Legacy AC       | Action                  | Scope                  |
|--------|----------------------------|-----------------|-------------------------|------------------------|
| GET    | /api/org/                  | reqSignedIn     | `orgs:read`             | `orgs:id:{OrgID}`      |
| GET    | /api/org/quotas            | reqSignedIn     | `orgs.quotas:read`      | `orgs:id:{OrgID}`      |
| GET    | /api/org/preferences       | reqOrgAdmin     | `orgs.preferences:read` | `orgs:id:{OrgID}`      |
| GET    | /api/orgs/:orgId           | reqGrafanaAdmin | `orgs:read`             | `orgs:id:{:orgId}`     |
| GET    | /api/orgs/:orgId/quotas    | reqGrafanaAdmin | `orgs.quotas:read`      | `orgs:id:{:orgId}`     |
| GET    | /api/orgs?perpage=x&page=y | reqGrafanaAdmin | `orgs:read`             | `orgs:*`               |
| GET    | /api/orgs/name/:orgName    | reqGrafanaAdmin | `orgs:read`             | `orgs:name:{:orgName}` |

I used this PR to correct a little bug I found on the `GET /api/org/quotas`, the `orgID` wasn't provided to the handler, resulting in `orgID: 0` quotas being displayed.

**Which issue(s) this PR fixes**:

https://github.com/grafana/grafana-enterprise/issues/1550

**Special notes for your reviewer**:
I won't merge this until I create the associated roles :slightly_smiling_face:

gamab committed:
AccessControl: Add FGAC to orgs endpoints

gamab committed:
Suggestion: Use singular for scopes

Co-authored-by: Karl Persson <kalle.persson@grafana.com>

gamab committed:
Suggestion: singular scopes

IevaVasiljeva commented on pkg/api/quota_test.go:
```diff
@@ -0,0 +1,125 @@
+	_, err = hs.SQLStore.CreateOrgWithMember("TestOrg2", testUserID)
+	require.NoError(t, err)
+
+	t.Run("AccessControl allows viewing another org quotas with correct permissions", func(t *testing.T) {
```
Might be worth adding a test with exact permissions here as well (read only org 2).

IevaVasiljeva commented on pkg/api/quota.go:
```diff
@@ -1,20 +1,29 @@
+	query := models.GetOrgQuotasQuery{OrgId: orgID}
 
-	if err := bus.DispatchCtx(c.Req.Context(), &query); err != nil {
+	if err := hs.SQLStore.GetOrgQuotas(c.Req.Context(), &query); err != nil {
```
What's the difference between importing `sqlstore` and calling `GetOrgQuotas` directly similarly as you do in `pkg/api/org.go` and using `SQLStore` field from `HTTPServer` to call its `GetOrgQuotas`?

gamab replied:
I'm replacing by the method that was registered on the bus to handle the `GetOrgQuotasQuery`. `GetOrgQuotas` is a method of the `SQLStore` pointer receiver.
On the other hand `GetOrgById` and `SearchOrgs` are public methods of the sqlstore package.

In general, I think we are moving towards using methods instead of public functions relying on package variables (safer). I won't refactor `SQLStore` here :smile:

IevaVasiljeva replied:
Thanks for the context! All makes sense. And please don't refactor `SQLStore`, that wasn't my intention at all 😆 I was just curious about the difference, as the action carried out in both places seemed pretty similar (retrieve some org related info from the db).

IevaVasiljeva reviewed (commented):
Overall looks good! Some questions and comments, but most of them are around tests.

gamab committed:
Suggestion: correct copy paste

Co-authored-by: Ieva <ieva.vasiljeva@grafana.com>

Fit reality with feature toggle

fix linting

Update common_test not to use macaron

Add a new set of permissions for preferences and quotas

IevaVasiljeva reviewed (approved):
Really great work on adding a neat test setup and extending tests in general! 🦸 

Looks good to me otherwise as well, just one small question regarding the new actions and fixed/built-in roles.

gamab committed:
AccessControl: Add FGAC permissions to org update endpoint (#39699)

* AccessControl: Add FGAC permissions to org update endpoint

gamab committed:
AccessControl: Add FGAC permissions to create and delete orgs (#40156)

* AccessControl: Add FGAC permissions to create and delete orgs

Co-authored-by: Ieva <ieva.vasiljeva@grafana.com>

gamab committed:
Merge branch 'main' into gamab/ac/orgs
